### DISCUSSION TITLE
Including common code in a function.

### DISCUSSION THREAD

**Discussion by 2vtech** [vercel/community#664] [Help] [answered]
I cannot figure out how to include code in a separate file into a function. For example, I have common code in a _share.js file I want to use in my /api/fun.js file. Import doesn't seem to work? What am I missing?

**Comment by wiiiimm** [comment]
importing should work. Can you share your code / repo here?

**Comment by 2vtech** [comment, marked as answer]
/api/date.js:

  import {TESTVAL} from '_test.js';
  
  console.log(`TESTVAL = ${TESTVAL}`);
  
  export default (req, res) => {
    const date = new Date().toString();
    res.status(200).send(date);
  }

/api/_test.js:
  export const TESTVAL = 10;

Error from running "vercel dev":

  Cannot find module '_test.js'
  Require stack:
  - C:\DevWork\2ValleyTech\eProdify\api\date.js
  - C:\Users\gregd\AppData\Roaming\npm\node_modules\vercel\node_modules\@vercel\node-bridge\launcher.js
  - C:\Users\gregd\AppData\Roaming\npm\node_modules\vercel\node_modules\@vercel\node\dist\dev-server.js
  Did you forget to add it to "dependencies" in `package.json`?
  Error! Failed to complete request to /api/date: Error: read ECONNRESET

**Reply by 2vtech** [reply]
I'm using Svelte if that matters.

**Reply by TooTallNate** [reply]
You need to use `./_test.js`, _not_ `_test.js` (note the `./` part):

```diff
- import {TESTVAL} from '_test.js';
+ import {TESTVAL} from './_test.js';
```

**Reply by 2vtech** [reply]
Just saw your reply. Thanks, my bad.

**Comment by 2vtech** [comment]
Any progress on this? Am I doing something wrong? This is really kinda a show-stopper for me.

**Reply by leerob** [reply]
Can you add a reproduction here? I am not able to reproduce.

**Comment by 2vtech** [comment]
I just deployed /api/date (the code above) and got this:

[GET] /api/date
22:12:46:05
2022-06-22T05:12:46.248Z	undefined	ERROR	Cannot find module '_test.js'
Require stack:
- /var/task/api/date.js
2022-06-22T05:12:46.249Z	undefined	ERROR	Did you forget to add it to "dependencies" in `package.json`?
RequestId: 2bd5f576-211a-4617-a6dd-98827580858b Error: Runtime exited with error: exit status 1
Runtime.ExitError


What else do you need to debug this?

**Comment by 2vtech** [comment]
The browser show this:

This Serverless Function has crashed.

Your connection is working correctly.

Vercel is working correctly.

500: INTERNAL_SERVER_ERROR
Code: FUNCTION_INVOCATION_FAILED
ID: pdx1::nl2wd-1655874766255-f4387958043c

If you are a visitor, contact the website owner or try again later.
If you are the owner, [learn how to fix the error](https://vercel.com/docs/error/application/FUNCTION_INVOCATION_FAILED) and [check the logs](https://www.eprodify.com/_logs).

**Comment by 2vtech** [comment]
Is there something I need to put into vercel.json to make this work?

I tried adding this, but it failed:

"functions": {
    "api/auth.js": {
        "includeFiles": "_db.js"
    }
  },

**Comment by 2vtech** [comment]
Fixed my issue: forgot to add relative path to filename, e.g. "./_test.js".
